pinot-core/src/main/java/org/apache/pinot/core/common/IntermediateStageBlockValSet.java

```diff
@@ -80,37 +83,37 @@ public int[] getDictionaryIdsSV() {
 
   @Override
   public int[] getIntValuesSV() {
-    return DataBlockUtils.extractIntValuesForColumn(_dataBlock, _index);
+    return DataBlockUtils.extractIntValuesForColumn(_dataBlock, _colIndex, _filterArgIdx);
```

Comment:
i dont know if this is the best API in DataBlockUtils to extract this. alternative is to extract the filter booleans and the IntValues separately, but i cannot make this change individually without changing how agg functions work.

Reply:
Is it possible to wrap a `FilteredTransferableBlock` on top of the `TransferableBlock` for `AggregateOperator`?

Reply:
Feeling shouldn't use these many if-else checks